nx/lib/nx/lin_alg.ex

```diff
@@ -38,31 +38,51 @@ defmodule Nx.LinAlg do
       >
   """
   defn adjoint(t, opts \\ []) do
-    case Nx.type(t) do
+    tensor = Nx.to_tensor(t)
+    opts = adjoint_opts(tensor.shape, opts)
+
+    case Nx.type(tensor) do
       {:c, _} ->
-        t |> Nx.transpose(opts) |> Nx.conjugate()
+        tensor |> Nx.transpose(opts) |> Nx.conjugate()
 
       _ ->
-        Nx.transpose(t, opts)
+        Nx.transpose(tensor, opts)
+    end
+  end
+
+  deftransformp adjoint_opts(shape, opts) do
+    rank = tuple_size(shape)
+
+    if rank > 2 do
+      axes = Enum.concat(0..(rank - 3), [rank - 1, rank - 2])
+      Keyword.put(opts, :axes, axes)
```

Comment:
This might make things confusing if the axes option was passed by the user

Reply:
Although it seems no options are actually documented, so perhaps we can remove the ability to pass any options as input to adjoint

Reply:
I did remove it, not sure why it was there in the first place